server/src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -822,6 +831,65 @@ public void setCapabilities(Set<ConsumerCapability> capabilities) {
         }
     }
 
+    /**
+     * Returns true if the consumer has the specified capability, or false otherwise.
+     *
+     * @param capability the capability we want to check for.
+     * @return true if the consumer has the capability, or false otherwise.
+     */
+    public boolean isCapable(String capability) {
+        if (this.getCapabilities() == null || capability == null) {
+            return false;
+        }
+
+        return this.getCapabilities().stream()
+            .map(ConsumerCapability::getName)
+            .anyMatch(capability::equals);
+    }
+
+    /**
+     * Was this consumer created in the last 24 hours?
+     *
+     * @return boolean was created in the last 24 hours
+     */
+    public boolean isNewborn() {
+        if (getCreated() == null) {
+            return false;
+        }
+        Instant oneDayFromRegistration = getCreated().toInstant().plus(Duration.ofDays(1));
+        return Instant.now().isBefore(oneDayFromRegistration);
+    }
+
+    /**
+     * Returns true if the consumer's architecture matches that of the input.
+     *
+     * @param productArch The architecture attribute value of a product.
+     * @return true if the architecture matches, false otherwise.
+     */
+    public boolean architectureMatches(String productArch) {
+
+        if (productArch != null) {
+            Set<String> supportedArches =
+                org.candlepin.util.Util.asSet(productArch.toUpperCase().trim().split("\\s*,\\s*"));
```

Comment:
This doesn't need to be fully qualified if this lives here

Reply:
You're right, forgot about that! Done